cmds/core/df/df.go

```diff
@@ -116,31 +130,42 @@ func DiskUsage(mnt *Mount) {
 	mnt.Used = (fs.Blocks - fs.Bfree) * uint64(fs.Bsize) / units
 	pct := float64((fs.Blocks - fs.Bfree)) * 100 / float64(fs.Blocks)
 	mnt.PCT = uint8(math.Ceil(pct))
+	return nil
 }
 
 // SetUnits takes the command line flags and configures
 // the correct units used to calculate display values
-func SetUnits() {
-	if *inKB && *inMB {
-		log.Fatal("options -k and -m are mutually exclusive")
+func SetUnits(inKB, inMB bool) error {
+	if inKB && inMB {
+		return errKMExclusiv
 	}
-	if *inMB {
+	if inMB {
 		units = MB
 	} else {
 		units = KB
 	}
+	return nil
 }
 
-func df() {
-	SetUnits()
-	mounts, _ := mountinfo()
+func df(args []string, fargs flags, w io.Writer) error {
```

Comment:
I would still prefer to see, as in the other code, this:
df(w io.Writer, fargs flags, args ...string)

Reply:
done